internal/p2p/router.go

```diff
@@ -303,47 +307,73 @@ type ChannelCreator func(context.Context, *ChannelDescriptor) (Channel, error)
 // wrapper message. The caller may provide a size to make the channel buffered,
 // which internally makes the inbound, outbound, and error channel buffered.
 func (r *Router) OpenChannel(ctx context.Context, chDesc *ChannelDescriptor) (Channel, error) {
-	r.legacy.channelMtx.Lock()
-	defer r.legacy.channelMtx.Unlock()
+	switch {
+	case r.options.UseLibP2P:
+		info := r.nodeInfoProducer()
+		ch, err := NewLibP2PChannel(info.Network, chDesc, r.options.NetworkPubSub, r.options.NetworkHost)
```

Comment:
Is there a reason to have the pubsub and host under both `options` and `network`?

Reply:
I think eventually we should not stash the options struct in the router, but that day is not today.